pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -58,6 +59,10 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, log *logrus.
 	isCreate := doc == nil
 
 	if isCreate {
+		err := f.validateVersion(installVersion)
```

Comment:
we should also validate during a PATCH request that the clusterProfile version is not modified